src/olympia/api/paginator.py

```diff
@@ -21,6 +21,8 @@ def validate_number(self, number):
             raise PageNotAnInteger('That page number is not an integer')
         if number < 1:
             raise EmptyPage('That page number is less than 1')
+        elif number > 100000:
```

Comment:
`index.max_result_window` is interesting but does not exist on the version we're using, though.